### DISCUSSION TITLE
Documentation For Debug Visuals (red bands)

### DISCUSSION THREAD

**Discussion by jimbz** [google/filament#3543] [Q&A]
Using a debug Android build of Filament and under some workloads I see one, sometimes two red bands on the left side of the rendered frames. It looks to me like it is a signal of something going wrong but I couldn't find where it happens nor what it means.

Would anyone have any pointer?

**Comment by prideout** [comment]
In debug builds, the GL backend clears discarded buffers to red in its implementation of `beginRenderPass`.  Let us know if you think this is might be a real bug; you might to check if the issue persists after toggling dynamic resolution and other various features in `View`.

**Reply by jimbz** [reply]
Thanks for the answer. I am aware of the clear to red features but it doesn't look like it to me. Here are screenshots of what I mean. I had to hide the actual scene with zero alpha as I am not able to share it, but it seems to be dependent on what is being displayed. It still happens but to a lesser extent with dynamic resolution enabled, which seems to confirm it might be load-related.

![Screenshot_20210217-091322_PHIAR AR Demo - TomTom](https://user-images.githubusercontent.com/728090/108241840-1cfa3e00-7101-11eb-9949-fa454f5f7382.jpg)

![Screenshot_20210217-092118_PHIAR AR Demo - TomTom](https://user-images.githubusercontent.com/728090/108242321-a447b180-7101-11eb-9854-7f68b93c47c2.jpg)

**Reply by jimbz** [reply]
This was tested on a Samsung S10 with Qualcomm Snapdragon 855 and Android 10. I checked the developer options for some GPU debugging toggles that might be enabled but didn't find anything.

**Reply by jimbz** [reply]
@prideout is it something you've seen before?

**Reply by prideout** [reply]
I haven't seen it before.  You said it only occurs in debug config, are you running into an actual issue with release builds?

**Reply by jimbz** [reply]
Mmm I a bit puzzled 🤔. Release builds behave as expected, just with less verbose logs and no red bars... This happens with the OpenGL backend. I can try with the Vulkan backend next.
